### PR TITLE
NF 272 Submit Amend and Confirmation Page display

### PR DESCRIPTION
**Pull request by njmoorewitchcat**

### CODE REVIEW

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimService.scala [hunk 1]
```diff
@@ -44,4 +47,18 @@ class CreateClaimService @Inject() (connector: NIDACConnector) {
...
   }
 
+  def amendClaim(amendClaim: AmendClaim)(implicit hc: HeaderCarrier): Future[AmendClaimResponse] = {
+
+    val correlationId = hc.requestId.map(_.value).getOrElse(UUID.randomUUID().toString)
+    val eisRequest: EISAmendCaseRequest = EISAmendCaseRequest(
+      AcknowledgementReference = correlationId.replace("-", "").takeRight(acknowledgementReferenceMaxLength),
+      ApplicationType = "NIDAC",
+      OriginatingSystem = "Digital",
+      Content = EISAmendCaseRequest.Content(amendClaim)
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimService.scala:57]: Nit: Consider constants for these two - especially as they are used in same class?

**Review by SteveSugden — DISMISSED** [review]
LGTM.   If you wanted to you could maybe add a `EISAmendCaseRequestValidationSpec` to match the "create" one - not that there is much to validate in the amend case schema.

**Review by rorticus0502 — COMMENTED** [review]
LGTM
Quite large in scope so best to let Steve and or Will approve it

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/models/eis/EISAmendCaseRequest.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ */
+
+package uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.models.eis
+
+import play.api.libs.json.{Format, Json}
+import uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.models.amend.AmendClaim
+
+/**
+  * Create specified case in the PEGA system.
+  * Based on spec "CSG_NIDAC_AutoCreateCase_API_Spec_V0.2.docx"  (NOTE: PEGA spec)
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/models/eis/EISAmendCaseRequest.scala:24]: OBSERVATION: This comment references the *create* case schema - should probably reference the PEGA *update* case schema (which is now in this repo following the addition of schema tests)

Comment (SteveSugden) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/models/eis/EISAmendCaseRequest.scala:24]: Good spot.  I actually didn't copy over the Word doc - just the JSON schema.   Maybe update the comment to reflect that here and in EIS create case?

Comment (njmoorewitchcat) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/models/eis/EISAmendCaseRequest.scala:24]: Will fix these now

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimService.scala (hunk 1 above)
Comment (njmoorewitchcat) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimService.scala:57]: Will add the test and pull those out as constants

File: conf/messages [hunk 3]
```diff
@@ -277,3 +277,6 @@ amend.check_answers.information.attach_more_documents.true = Yes
 amend.check_answers.information.uploadedDocuments = Additional documents
 amend.check_answers.submit.heading = Submit your declaration
 amend.check_answers.submit.declaration = Before you proceed, ensure that you have fully reviewed the information you have provided above and ensure that it is correct.
+
+amend.confirmation.title = Information Sent
```
Comment (crosslandwa) [inline conf/messages:281]: CLARIFICATION: Is the capitalisation of `Sent` intentional/as per designs? The other `xxx.xxx.title` values in the messages file only capitalise the first word...

**Review by crosslandwa — APPROVED** [review]
LGTM - a couple of minor OBSERVATIONS/CLARIFICATIONS that I assume will be addressed in necessary (and ignored if i'm wrong)

File: conf/messages (hunk 3 above)
Comment (njmoorewitchcat) [reply conf/messages:281]: Nope - that's me - will fix now.

**Review by SteveSugden — APPROVED** [review]
Changes LGTM

**Review by crosslandwa — APPROVED** [review]

**Pull request merged**